provisionersdk/provisionertags.go

```diff
@@ -14,23 +14,27 @@ const (
 // If the scope is "user", the "owner" is changed to the user ID.
 // This is for user-scoped provisioner daemons, where users should
 // own their own operations.
-// Otherwise, the "owner" tag is always empty.
+// Otherwise, the "owner" tag is always an empty string.
+// NOTE: "owner" must NEVER be nil. Otherwise it will end up being
+// duplicated in the database, as idx_provisioner_daemons_name_owner_key
+// is a partial unique index that includes a JSON field.
 func MutateTags(userID uuid.UUID, tags map[string]string) map[string]string {
 	if tags == nil {
 		tags = map[string]string{}
 	}
 	_, ok := tags[TagScope]
 	if !ok {
 		tags[TagScope] = ScopeOrganization
-		delete(tags, TagOwner)
+		tags[TagOwner] = ""
 	}
 	switch tags[TagScope] {
 	case ScopeUser:
 		tags[TagOwner] = userID.String()
 	case ScopeOrganization:
-		delete(tags, TagOwner)
+		tags[TagOwner] = ""
 	default:
 		tags[TagScope] = ScopeOrganization
+		tags[TagOwner] = ""
```

Comment:
review: calling this out here -- I had assumed that the index over `(name, tags["owner"])` would disallow multiple rows with `("foo", NULL)` but this is apparently not the case. However, it does disallow multiple rows with `("foo", "")`.

I updated to use `MutateProvisionerTags` wherever possible when inserting a provisioner daemon.

Reply:
@mafredri helpfully pointed out that we can use `COALESCE` in the index to ensure that an absent value is treated the same as an empty string. 🎉 Adding a migration for this now.